lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -81,12 +81,42 @@ openModuleFiles(StringRef DirName, StringRef ModuleFilename,
   return std::error_code();
 }
 
+static void addDiagnosticInfoForArchitectureMismatch(ASTContext &ctx,
+                             SourceLoc sourceLocation,
+                             llvm::SmallString<64> moduleName,
+                             llvm::SmallString<16> archName,
+                             llvm::SmallString<128> directoryPath) {
+
+  std::error_code errorCode;
+  llvm::Twine twineDirPath(directoryPath);
+  llvm::sys::fs::directory_iterator directoryIterator(twineDirPath,
+                                                      errorCode,
+                                                      true);
+  llvm::sys::fs::directory_iterator endIterator;
+
+  if(errorCode) {
+      return;
+    }
+
+  std::string foundArchs;
+  for (; directoryIterator != endIterator; directoryIterator.increment(errorCode)) {
+    auto entry = *directoryIterator;
+    llvm::StringRef filePath(entry.path());
+    foundArchs = foundArchs + (foundArchs.length() > 0 ? ", " : "")
```

Comment:
A swift module contains swiftdoc files as well.  This will pick those up and double-diagnose.

Reply:
True. Let me add a check for that !! And update the test accordingly.